engine/src/main/java/org/camunda/bpm/engine/impl/util/ImmutablePair.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.util;
+
+import java.io.Serializable;
+import java.util.Map.Entry;
+
+import java.util.Objects;
+
+/**
+ * Immutable representation of a 2-tuple of elements.
+ *
+ * Although the implementation is immutable, there is no restriction on the
+ * objects that may be stored. If mutable objects are stored in the pair, then
+ * the pair itself effectively becomes mutable.
+ *
+ * ThreadSafe if both paired objects are thread-safe
+ *
+ * @param <L>
+ *          the type of the left element
+ * @param <R>
+ *          the type of the right element
+ */
+public class ImmutablePair<L, R> implements Entry<L, R>, Serializable, Comparable<ImmutablePair<L, R>> {
+
+  /** Serialization version */
+  private static final long serialVersionUID = -7043970803192830955L;
+
+  private L left;
```

Comment:
Modifier `private` is not allowed for fields and methods. Please also see line 44 & 117

Reply:
It's fine for the `serialVersionUID`, right?

Reply:
Yes, as far as I know it's not possible to override static fields anyways.